src/ApcuRateLimiter.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RateLimit;
+
+use RateLimit\Exception\CannotUseRateLimiter;
+use RateLimit\Exception\LimitExceeded;
+
+final class ApcuRateLimiter implements RateLimiter, SilentRateLimiter
+{
+    /** @var string */
+    private $keyPrefix;
+
+    public function __construct(string $keyPrefix = '')
+    {
+        if (!\extension_loaded('apcu') || \ini_get('apc.enabled') === '0') {
+            throw new CannotUseRateLimiter('APCu extension is not loaded or not enabled.');
+        }
+
+        if (\ini_get('apc.use_request_time') === '1') {
+            throw new CannotUseRateLimiter('APCu ini configuration "apc.use_request_time" should be set to "0".');
+        }
+
+        $this->keyPrefix = $keyPrefix;
+    }
+
+    public function limit(string $identifier, Rate $rate): void
+    {
+        $limitKey = $this->limitKey($identifier, $rate->getInterval());
+
+        $current = $this->getCurrent($limitKey);
+        if ($current >= $rate->getOperations()) {
+            throw LimitExceeded::for($identifier, $rate);
+        }
+
+        $this->updateCounter($limitKey, $rate->getInterval());
+    }
+
+    public function limitSilently(string $identifier, Rate $rate): Status
+    {
+        $interval = $rate->getInterval();
+        $limitKey = $this->limitKey($identifier, $interval);
+        $timeKey = $this->timeKey($identifier, $interval);
+
+        $current = $this->getCurrent($limitKey);
+        if ($current <= $rate->getOperations()) {
+            $current = $this->updateCounterAndTime($limitKey, $timeKey, $interval);
+        }
+
+        return Status::from(
+            \sprintf('%s%s', $this->keyPrefix, $identifier),
```

Comment:
`keyPrefix` is not needed here. `Status` is storage-agnostic, it provides rate limiting information about the original identifier (IP address, user ID, API key) regardless of whether information is stored in Redis, APC or some other backend. `keyPrefix` is an implementation detail of this particular backend.